pkg/remote/google/repository/asset.go

```diff
@@ -75,102 +76,131 @@ func NewAssetRepository(client *asset.Client, config config.GCPTerraformConfig,
 }
 
 func (s assetRepository) listAllResources(ty string) ([]*assetpb.Asset, error) {
-	req := &assetpb.ListAssetsRequest{
-		Parent:      fmt.Sprintf("projects/%s", s.config.Project),
-		ContentType: assetpb.ContentType_RESOURCE,
-		AssetTypes: []string{
-			cloudFunctionsFunction,
-			bigtableInstanceAssetType,
-			bigtableTableAssetType,
-			sqlDatabaseInstanceAssetType,
-			computeGlobalAddressAssetType,
-			nodeGroupAssetType,
-		},
-	}
-	var results []*assetpb.Asset
 
-	cacheKey := "listAllResources"
-	cachedResults := s.cache.GetAndLock(cacheKey)
-	defer s.cache.Unlock(cacheKey)
-	if cachedResults != nil {
-		results = cachedResults.([]*assetpb.Asset)
-	}
+	filteredResults := []*assetpb.Asset{}
+	var errorString string
+
+	for _, scope := range s.config.Scopes {
+		cacheKey := fmt.Sprintf("listAllResources_%s", scope)
+		cachedResults := s.cache.GetAndLock(cacheKey)
+		defer s.cache.Unlock(cacheKey)
+
+		req := &assetpb.ListAssetsRequest{
+			Parent:      scope,
```

Comment:
I'm not sure that a folder is supported for `listAssets` call according to the documentation here : https://cloud.google.com/asset-inventory/docs/reference/rpc/google.cloud.asset.v1#google.cloud.asset.v1.ListAssetsRequest

Can you double check that ?

Reply:
Indeed. Ideally we should not use listAssets but only searchAssets as the main difference is that list takes a snapshot of how the assets were at a particular time, which could result in a diff from what it's actually present.

Reply:
Where did you get this info ? I asked this question [here](https://stackoverflow.com/questions/69648122/difference-between-listassets-and-searchallresources) a couple of month ago and I didn't get any answer.

Reply:
From [here](https://cloud.google.com/asset-inventory/docs/listing-assets) there are a couple of hints about it:

```
Listing assets might not meet the performance requirements for large scale customers. If this is the case for you and calling list API encounters timeout, it's recommended to use export API instead.
```

Also in [the API](https://cloud.google.com/asset-inventory/docs/reference/rest/v1/assets) it's mentioned the purpose is to list resources from a specific timeframe

```
Lists assets with time and resource types and returns paged results in response.
readTime:  Timestamp to take an asset snapshot.  If not specified, the current time will be used. Due to delays in resource data collection and indexing, there is a volatile window during which running the same query may get different results.
```

While the search is more appropriate for our intent:

```
The Cloud Asset API allows you to use a custom query language to query resource metadata on a project, folder, or organization.
Method: searchAllResources. Searches all Cloud resources within the specified scope, such as a project, folder, or organization. 
```